src/Collector/Finder/ResolvedNodeFinder.php

```diff
@@ -18,15 +21,43 @@ final class ResolvedNodeFinder
      */
     private array $collectedResolvedNodes = [];
 
-    public function __construct(CollectedDataNode $collectedDataNode)
+    /**
+     * @var array<string>
+     */
+    private array $analysedFiles = [];
+
+    /**
+     * @param string[] $existingTemplates
+     * @param LatteTemplateResolverInterface[] $latteTemplateResolvers
+     */
+    public function __construct(CollectedDataNode $collectedDataNode, array $existingTemplates, array $latteTemplateResolvers)
     {
+        // node resolvers
         $collectedResolvedNodes = ResolvedNodeCollector::loadData($collectedDataNode, CollectedResolvedNode::class);
+
+        // file resolvers
+        foreach ($existingTemplates as $templateFile) {
```

Comment:
I think we don't need this part, only LatteCustomTemplateResolverInterface. If we want to use existing templates we can implement it with LatteCustomTemplateResolverInterface. It solves my and also mabar's problem.

Reply:
Well I added it first and I think it might be most common use case. Also it reduces dependencies on kinf-of internal classes like AnalasedTemplatesRegistry - used to find latte files

Reply:
But I can remove it and keep jsut universal custom resolver

Reply:
if I'm right then yes please, remove it.

Reply:
REmoved. We can make that code part of some kind of AbstractFileResolver later